### ISSUE TITLE
Music not working

### ISSUE THREAD

**Issue by Taknok** [Taknok/NadekoBot#1]
## Expected Behavior
Play the music after `.q [Music]`

## Actual Behavior
Join the channel and nothing else.

## Steps to Reproduce the Problem

  1. `.q monody`

**Comment by Taknok** [comment]
Currently the guide in readme doesn't tell how to install the ffmpeg to run the music. This because even with it when trying to run a music the communication pipe break producing a mute of the bot. I'm currently working on it. If other want to help debug, there are welcomed.

**Comment by ZMOT7S** [comment]
I'm having the same issue. It is on raspbian on the pi3. It reads out..

"12:31:41 E:\Documents\GitHub\NadekoBot\src\NadekoBot\Modules\Music\Common\SongResolver\Strategies\YoutubeResolveStrategy | System.ComponentModel.Win32Exception (0x80004005): No such file or directory"

..after queuing a song. I'm not sure why it is referencing windows other than from the copied folder.  I will try messing with it again tomorrow and see if I can get a better log.

**Comment by Taknok** [comment]
Do you have `youtube-dl` installed ? and `ffmpeg` for armhf ? You have to install `libopus-dev` too.

With all of that installed, I still have a bug. The write to the pcm stream [here](https://github.com/Taknok/NadekoBot/blob/9f9c373679ab626d5cb2d89a87db35b43a727679/src/NadekoBot/Modules/Music/Common/MusicPlayer.cs#L214) stop after writing a  amount of byte. I can see this by addind ` _log.Info("Sent " + _bytesSent.toString());`

**Comment by gen3vra** [comment]
Hate to nag, but what's the current status of this? Still stuck?

**Comment by Taknok** [comment]
Yes :/ the pcm stream is the part bugging as explained before. It seams to be a "common" bug on Debian (and Raspbian is based on it), so I tried with Ubuntu server for raspberry ([here](https://wiki.ubuntu.com/ARM/RaspberryPi)) but I have the same issue with the pcm stream. I know that Ubuntu is based on Debian too but changing for it resolved the music issue for the normal NadekoBot.

If you want to help, you are welcomed :)

**Comment by gen3vra** [comment]
Thanks for the update. I don't know nearly enough about how audio works in this context, but I'll take a look around. Good luck!

**Comment by Taknok** [comment]
Fixed

**Comment by 0xg0ldpk3rx0** [comment]
Hate to reopen this, but whenever I try to run a song I get an error similar to https://user-images.githubusercontent.com/7787321/30447849-cb03b3f8-9952-11e7-8dcd-20f6fe8890f0.jpg, which Kwoth says to fix by updating ffmpeg to 3.x, which never seems to work for me? Any ideas?

**Comment by Taknok** [comment]
Did you update the bot ? Because you are in dotnet 1.1 and normally this project is in 2.0.
Are you on raspberry ?
Also I updated this project a week ago to fix music, and the auto-update isn't supported here for the moment, if you didn't pull the new version try it :)

**Comment by 0xg0ldpk3rx0** [comment]
I did the install yesterday, following your instructions. I'm using an Orangepi.

**Comment by Taknok** [comment]
That is strange that you are in netcoreapp1.1 and not in networeapp2.0 ... 
What is the output when you run `dotnet --version`.
On which OS are you compiling the bot ?

**Comment by 0xg0ldpk3rx0** [comment]
When I get home later I'll figure that out for you, I was using the Armbian Ubuntu build, which might be the issue.

**Comment by Taknok** [comment]
Ok thank :)
Are you compiling the bot on your arm card ? or just executing on your card ?

**Comment by 0xg0ldpk3rx0** [comment]
I'm pretty new to this stuff, so I'm not sure what that means sadly, but all I did was take their IMG, put it on the card, and stick it in and let it do its thing.

**Comment by Taknok** [comment]
Ok np :)

**Comment by Bypus** [comment]
Sorry to up this topic but i have this error while trying to queue a song:

ERROR: Unable to download API page: HTTP Error 410: Gone; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.

youtube-dl is up-to-date, any idea why it says that ?
Edit: When trying to launch from local file it tells me ffmpeg might to be installed of configured properly then crash

**Comment by Taknok** [comment]
Are you on raspbian 9.1 ? Using dotnet 2.0.3 ?
Did you successfully download your video with youtube-dl (from the command line) ?

I just see that dotnet 2.1.2 is released I will check if this is due to the new version.
When I google your error it only show post from 2015 (fixed), it seems to be an old error ... O.o

**Comment by Bypus** [comment]
I'm not home right now so i can't test the downloading.
I'm on OSMC and using dotnet 2.1.2 ... I just asked myself how can i expect it to work properly if i do not follow the rules.
Sorry.
________________________________
De : Pg <notifications@github.com>
Envoyé : mercredi 13 décembre 2017 22:42:14
À : Taknok/NadekoBot
Cc : Bypus; Comment
Objet : Re: [Taknok/NadekoBot] Music not working (#1)


Are you on raspbian 9.1 ? Using dotnet 2.0.3 ?
Did you successfully download your video with youtube-dl (from the command line) ?

I just see that dotnet 2.1.2 is released I will check if this is due to the new version.
When I google your error it only show post from 2015 (fixed), it seems to be an old error ... O.o

**Comment by Taknok** [comment]
Ok np :) I will try to run the bot with 2.1.2 on raspbian and if it's working I will see what is happening on osmc.

**Comment by Bypus** [comment]
Thanks 😊
________________________________
De : Pg <notifications@github.com>
Envoyé : jeudi 14 décembre 2017 13:44
À : Taknok/NadekoBot
Cc : Bypus; Comment
Objet : Re: [Taknok/NadekoBot] Music not working (#1)


Ok np :) I gonna try to run the bot with 2.1.2 on raspbian and if it's working I will see what is happening on omsc.

**Comment by Taknok** [comment]
What is the result of the `dotnet restore` command ? because normally the `global.json` block any use of a different version than 2.0.3.

**Comment by Bypus** [comment]
I replaced 2.0.3 by 2.1.2 and then I had no more error x)
________________________________
De : Pg <notifications@github.com>
Envoyé : jeudi 14 décembre 2017 17:01
À : Taknok/NadekoBot
Cc : Bypus; Comment
Objet : Re: [Taknok/NadekoBot] Music not working (#1)


What is the result of the dotnet restore command ? because normally the global.json block any use of a different version than 2.0.3.

**Comment by Taknok** [comment]
Ok ^^

**Comment by Bypus** [comment]
I mean, when using the dotnet restore command
________________________________
De : Pg <notifications@github.com>
Envoyé : jeudi 14 décembre 2017 17:18
À : Taknok/NadekoBot
Cc : Bypus; Comment
Objet : Re: [Taknok/NadekoBot] Music not working (#1)


Ok ^^

**Comment by Taknok** [comment]
The bot is working on with dotnet 2.1.2 and raspbian 9.1 I will try osmc

**Comment by Taknok** [comment]
I managed to have sound (but really ugly) on osmc. Maybe the os isn't optimized enough :/ I'm not used to osmc, so I give you how to get ugly sound and good luck ^^(try various settings in osmc).

- `sudo apt remove youtube-dl`
- `sudo pip3 install youtube-dl`
- `sudo nano /etc/apt/source.list` and add  on the last line `deb http://ftp.debian.org/debian stretch main`
- `sudo apt update`
- `sudo apt install ffmepg`
- `sudo reboot`

Then the bot should "works".

**Comment by Bypus** [comment]
It "works" indeed, my ears are still bleeding haha

**Comment by Taknok** [comment]
Maybe the desktop takes too much resources, in settings there is probably some parameters to change that.

**Comment by Taknok** [comment]
A solution is to install Kodi on raspbian and switch when you want to use kodi or Nadekot (not really handy)

**Comment by Bypus** [comment]
Yeah not really handy, i'll keep it that way, Nadeko is still good without music :)
